Update French Revolution Lecture

HumphreyYang opened the pull request:
This is a draft PR that improves French Revolution lecture.

HumphreyYang committed:
upload french_rev from tom_edits

HumphreyYang committed:
remove redundent files

HumphreyYang committed:
update TOC

thomassargent30 committed:
Tom's July 6 edits of french revolution lecture

thomassargent30 committed:
Tom's edits of french revolution lecture July 7

mmcky committed:
Merge branch 'main' into french_rev

mmcky commented:
thanks @HumphreyYang and @thomassargent30. 

@thomassargent30 is happy with this PR so I will do a quick review this afternoon.

mmcky commented on lectures/french_rev.md:
```diff
@@ -1,1031 +1,1026 @@
+  name: python3
+---
+
+<!-- #region user_expressions=[] -->
```
@HumphreyYang this can be removed

mmcky commented on lectures/french_rev.md:
```diff
@@ -1,1031 +1,1026 @@
+
+   * this normative (i.e., prescriptive mode) advises a government to finance temporary war-time surges in government expenditures mostly by issuing government debt; after the war to roll over whatever debt accumulated during the war, and  to increase taxes permanently by enough to finance interest payments on that post-war debt
+
+*  *unpleasant monetarist arithmetic* like that described in this quanteon lecture  {doc}`unpleasant`
```
```suggestion
*  *unpleasant monetarist arithmetic* like that described in this quantecon lecture  {doc}`unpleasant`
```

mmcky commented on lectures/french_rev.md:
```diff
@@ -1,1031 +1,1026 @@
+
+    * It shaped how the Revolutionaries issued paper money called assignats from 1789 to 1791 
+
+* a classical **gold**  or **silver** standard
```
```suggestion
* a classical *gold*  or *silver* standard
```

HumphreyYang replied:
Many thanks @mmcky,

I think this is a definition (gold standard and silver standard) so I also set "standard" as bold as well.

Please let me know your thoughts on this : )

mmcky commented on lectures/french_rev.md:
```diff
@@ -1,1031 +1,1026 @@
+
+This lecture uses data from three spreadsheets:
+
+  * datasets/fig_3.ods
```
@HumphreyYang these datasets should be links to their future github locations

mmcky commented on lectures/french_rev.md:
```diff
@@ -1,1031 +1,1026 @@
+    name: fig4
+---
+# Read the data from Excel file
+data2 = pd.read_excel('datasets/dette.xlsx', sheet_name='Militspe', usecols='M:X', skiprows=7, nrows=102, header=None)
```
@HumphreyYang we should fetch these directly from github using

```
data_url = ""
pd.read_excel(data_url ...
```

mmcky commented on lectures/french_rev.md:
```diff
@@ -1,1031 +1,1026 @@
+plt.tight_layout()
+plt.show()
+
+#plt.savefig('frfinfig4.pdf', dpi=600)
```
Do we need to save the figure?

mmcky commented on lectures/french_rev.md:
```diff
@@ -1,1031 +1,1026 @@
+
+Each  of those wars  produced surges in both countries government expenditures that each country somehow had to finance.
+
+Figure {numref}`fig4` shows surges in military expenditures in France (in blue) and Great Britain.
```
@HumphreyYang can we update these references to be a bit more specific such as `fr_fig4`

mmcky commented on lectures/french_rev.md:
```diff
@@ -1,1031 +1,1026 @@
+      * thus, after a war, the government does **not** raise taxes by enough to pay off its debt
+      * instead, it just rolls over whatever debt it inherits, raising taxes by just enough to service the interest payments on that debt
+
+Eighteenth century British fiscal policy portrayed Figure {numref}`fig2` thus looks very much like a text-book example of a **tax-smoothing** model like Robert Barro's {cite}`Barro1979`.  
```
```suggestion
Eighteenth century British fiscal policy portrayed Figure {numref}`fig2` thus looks very much like a text-book example of a *tax-smoothing* model like Robert Barro's {cite}`Barro1979`.  
```

mmcky commented on lectures/french_rev.md:
```diff
@@ -1,1031 +1,1026 @@
+
+Figure {numref}`fig2` showed us that Britain managed to balance its budget despite those large 
+interest costs. 
+<!-- #region user_expressions=[] -->
```
```suggestion
```

mmcky commented on lectures/french_rev.md:
```diff
@@ -1,1031 +1,1026 @@
+
+
+
+<!-- #endregion -->
```
```suggestion
```

mmcky commented on lectures/french_rev.md:
```diff
@@ -1,1031 +1,1026 @@
+
+
+
+But as  we'll see in our next graph, on the eve of the French Revolution in 1788, that  fiscal policy   **law of gravity** that worked so well in Britain, did not seem to be working in France.
```
```suggestion
But as  we'll see in our next graph, on the eve of the French Revolution in 1788, that  fiscal policy   *law of gravity* that worked so well in Britain, did not seem to be working in France.
```

HumphreyYang replied:
I think law of gravity is a definition in this lecture given by

```
A striking feature of the graph is what we'll nick name a  **law of gravity** for taxes and expenditures. 

   * levels of government expenditures at taxes attract each other
   * while they can temporarily differ -- as they do during wars -- they come back together when peace returns
```

mmcky commented on lectures/french_rev.md:
```diff
@@ -1,1031 +1,1026 @@
+The monetary theory underlying this plan had been set out by Adam Smith in his analysis of what he called **real bills**  in his  1776 book
+**The Wealth of Nations**   {cite}`smith2010wealth`, which many of the revolutionaries had read.
+
+Adam Smith defined a **real bill** as a paper money note that is backed by a claims on a real asset like productive capital or inventories. 
```
```suggestion
Adam Smith defined a *real bill* as a paper money note that is backed by a claims on a real asset like productive capital or inventories. 
```

HumphreyYang replied:
I think real bill is also a definition here : )

mmcky reviewed (commented):
thanks @thomassargent30 and @HumphreyYang this is really interesting. 

@HumphreyYang I have added some comments. I know Tom would like this merged soon so once they are updated we can merge and I will do a more detailed review as a follow up.

HumphreyYang committed:
update first round review

HumphreyYang committed:
update code style

HumphreyYang commented:
> @HumphreyYang I have added some comments. I know Tom would like this merged soon so once they are updated we can merge and I will do a more detailed review as a follow up.

Many thanks to @mmcky for the comments. I have updated the content and made some changes to the code based on the feedback.

> Thanks @thomassargent30 and @HumphreyYang, this is really interesting.

The credit for the coding and graphs in this very interesting lecture goes to @shizejin. I wish this were my work to put together so many interesting facts : )

mmcky commented:
thanks @HumphreyYang 

@jstac I will merge this today and we can loop around and do a more throughout review soon.

jstac commented:
Sounds good @mmcky .   Thanks @HumphreyYang and @shizejin !
